builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -763,6 +826,60 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	// Read the bucket replication configuration
+	replication, err := s3conn.GetBucketReplication(&s3.GetBucketReplicationInput{
+		Bucket: aws.String(d.Id()),
+	})
+	if err != nil {
+		if awsError, ok := err.(awserr.RequestFailure); ok && awsError.StatusCode() != 404 {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] S3 Bucket: %s, read replication configuration: %v", d.Id(), replication)
+	if r := replication.ReplicationConfiguration; r != nil {
```

Comment:
The only comment I have here is to extract this into a flatten function - this will allow the code to be a little more concise in this area

I suggest something like this:

```
if replication.ReplicationConfiguration != nil {
  if err := d.Set("replication_configuration", flattenAwsS3BucketReplicationConfiguration(replication.ReplicationConfiguration)); err != nil {
		log.Printf("[DEBUG] Error setting replication configuration: %s", err)
		return err
	}
}
````